src/serde/read_cache_lookup.rs

```diff
@@ -29,10 +30,10 @@ pub struct ReadCacheLookup {
     /// the tree hashes of the contents on the left and right
     read_stack: Vec<(Bytes32, Bytes32)>,
 
-    count: HashMap<Bytes32, u32>,
+    count: HashMap<Bytes32, u32, Bytes32Hasher>,
```

Comment:
wouldn't it make more sense to `impl Hash for Bytes32`?
This also highlights a potential attack, if the hashing is consistent, hash collisions can be forced by an attacker, causing the hash table to have O(n) lookups. My (perhaps optimistic) guess is that the default hasher mixes in some secret session key to make it impossible to attack.

Reply:
I was a bit confused. If you want a cheap hash function, you need to implement the `Hasher` yourself, of course. However, this hasher doesn't really have anything to do with `Bytes32`, so I think the name is a bit misleading. `XorHasher` or something would make more sense I think.